urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/resource/HypertextAccessHandler.java

```diff
@@ -0,0 +1,614 @@
+	    
+	    throw new MissingParamException( "Cannot complete authorization process. The response misses some required essentials." );
+	    
+	}
+
+
+	// The passed nonce and the local nonce MUST BE EQUAL! (the same counts for the realm?!)
+	// ...
+	
+
+	// ...
+	// Here: all credentials are present.
+
+	
+	this.logger.log( Level.INFO,
+			 getClass().getName() + ".checkValidUser_digestAuthentication_(...)",
+			 "challengeMap=" + challengeMap );
+
+	return checkValidUser_digestAuthentication_processCredentials( requestURI, headers, htaccess, authMethod, authUserFile, 
+								       additionalSettings, optionalReturnSettings,
+								       challengeMap,
+								       chl_algorithm,
+								       chl_realm,
+								       chl_response,
+								       chl_uri,
+								       chl_username,
+								       chl_nonce
+								       );
+
+    }
+
+
+    private boolean checkValidUser_digestAuthentication_processCredentials(  URI requestURI,
+									     HTTPHeaders headers,
+									     HypertextAccessFile htaccess,
+									     String authMethod,
+									     File authUserFile,
+									     Map<String,BasicType> additionalSettings,
+									     Map<String,BasicType> optionalReturnSettings,
+
+									     Map<String,String> challengeMap,
+
+									     String chl_algorithm,
+									     String chl_realm,
+									     String chl_response,
+									     String chl_uri,
+									     String chl_username,
+									     String chl_nonce
+									     ) 
+	throws IOException,
+	       MissingParamException,
+	       ConfigurationException {
+
+	
+	// Calculate the final MD5 checksum (hopefully the same as specified in the htaccess file).
+	// See: http://blog.smartbear.com/software-quality/bid/169684/How-to-Use-Digest-HTTP-Authentication-in-soapUI
+	// ------------------------------------------------------
+	// [...]
+	// In the first version ( RFC 2069 ) the algorithm is:
+	//
+	// A1 = username:realm:password
+	// A2 = method:digestURI
+	// HA1 = MD5( A1 )
+	// HA2 = MD5( A2 )
+	// response = MD5( HA1:nonce:HA2 )
+	// [...]
+	// ------------------------------------------------------
+	// For more details see 
+	//  - http://tools.ietf.org/html/rfc2617
+	// and
+	//  - http://tools.ietf.org/html/rfc2069
+
+
+
+	char[] filePassData = null;
+	try {
+	    filePassData = HypertextPasswordFile.getPasswordData( authUserFile, 
+								  chl_username, 
+								  this.strictMode );
+	    // Found in file?
+	    if( filePassData == null ) 
+		return false;       // Not in the user list
+
+	    // Password data exists -> store into returnSettings
+	    this.storeAuthUserSettings( filePassData, optionalReturnSettings );
+	    
+	    
+	    String[] filePassData_split = null;
+	    //if( !MD5.isIKRSMD5(new String(filePassData)) ) {
+	    try {
+
+		filePassData_split = MD5.splitIKRSMD5( new String(filePassData) );
+
+
+	    } catch( NumberFormatException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 520.